packages/contracts/src/contracts/current/protocol/Exchange/MixinExchangeCore.sol

```diff
@@ -119,6 +128,12 @@ contract MixinExchangeCore is
             return 0;
         }
 
+        // Validate order is not cancelled
+        if (order.salt < makerEpoch[order.makerAddress]) {
```

Comment:
The default value of a makerEpoch is `0`, so this test has to be `<` otherwise an uninitialized makerEpoch will result in a false positive. We could change the effect of `cancelOrdersBefore` to set the makerEpoch to `salt + 1`, which would allow users to input the salt of their latest order and cancel all orders up to that point, inclusively. In this case, we may also want to change `cancelOrdersBefore` to something like `cancelOrdersUpTo`. Any input on naming?

Reply:
@abandeali1 - this change has been applied